File: 2023/01/b.py
import re


def processa_linha(linha: str)-> int:
    dezenas = 10000
    dval = 0
    unidades = -1
    uval = 0
    nums = [
        "0",
        "1",
        "2",
        "3",
        "4",
        "5",
        "6",
        "7",
        "8",
        "9",
        "zero",
        "one",
        "two",
        "three",
        "four",
        "five",
        "six",
        "seven",
        "eight",
        "nine",
    ]

    for num in nums:
        for m in re.finditer(num, linha):
            if m.start() < dezenas:
                print("dez")
                dezenas = m.start()
                dval = nums.index(m.group()) % 10
            if m.start() > unidades:
                print("uni")
                unidades = m.start()
                uval = nums.index(m.group()) % 10
            print(m.start(), m.end(), m.group(), unidades, dezenas, dval, uval)
    return 10 * dval + uval


inpf = open("input", "r")

total = 0
for line in inpf:
    print(line.strip())
    _a = processa_linha(line.strip())
    total += _a
    print(_a, total)

File: 2023/01/b.jl
function processa_linha(line, nums, renums)
    rline = reverse(line)
    ff = line[findfirst(Regex(renums), line)]
    fl = reverse(rline[findfirst(Regex(reverse(renums)), rline)])
    d = (findfirst(item -> item == ff, nums) - 1) % 10
    u = (findfirst(item -> item == fl, nums) - 1) % 10

    return d * 10 + u
end

function parta()
    lines = readlines(Base.source_dir() * "/input")
	nums = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
    renums = "0|1|2|3|4|5|6|7|8|9"
    total = 0
    for line in lines
        total += processa_linha(line, nums, renums)
    end
    println("-- Total A ", total)
end

function partb()
    lines = readlines(Base.source_dir() * "/input")
	nums = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",]
    renums = "0|1|2|3|4|5|6|7|8|9|zero|one|two|three|four|five|six|seven|eight|nine"
    total = 0
    for line in lines
        total += processa_linha(line, nums, renums)
    end
    println("-- Total B ", total)
end

parta()
partb()